enhancements/results-forwarding.md

```diff
@@ -0,0 +1,379 @@
+
+* Address the aforementioned limitations with Etcd by choosing an alternative
+  store. e.g. a relational database or a cloud-managed database.
+
+The Compliance Operator also stores raw results (currently ARF) in a PVC by default.
+While this is useful, it has proven problematic in multiple deployments given that
+not everybody has access to persistent storage in their clusters.
+
+ARF files contain a lot of information about the state of the system where the
+compliance scan was effectuated. We could take the view that this is "evidence"
+gathered for the system in check. With this in mind, we can generalize and
+expand the concept to allow for forwarding this evidence to an alternative
+object store, e.g. S3.
+
+In the future, we could expand the "evidence" gathered, e.g. for Kubernetes
+compliance scans we could simply compress the gathered objects the
+`api-resource-collector` downloads and send that too.
+
+### Goals
+
+* Provide a generic way to implement compliance result forwarding for Compliance Operator.
+  This should be flexible enough to accommodate the current implementation (CRDs and PVCs),
+  while allowing for custom forwarder implementations.
+
+* Implement and define a stable API for forwarder implementations.
+
+* Implement result/remediation CRDs and ARF PVC as reference implementations of the forwarding
+  model.
+
+### Non-Goals
+
+* Deprecate the current CRD/PVC approach: There is still a use-case for this.
+
+* Expand the "evidence" that's currently gathered by the Compliance Operator. This is
+  needed but should be done as part of a separate initiative.
+
+## Proposal
+
+In short, the proposal is to have Compliance Operator *always* forward Compliance results (suggested
+remediations being part of this), as well as evidence (e.g. ARF results). This will require
+changing the **aggregator** to always forward via a GRPC to a configured implementation. The
+**resultserver** then becomes subject to the evidence forwarding provider and will be renamed
+**evidence-persistor**.
+
+Forwarder implementations would be configured via the `ScanSetting` and they'd be called "providers".
+For backwards compatibility, we'd have defaults that would point to the current mode of operation
+of the Compliance Operator. e.g. a `pvc` provider for evidence storage.
+
+For more detail on the proposed API, see the *API Extensions* section.
+
+### User Stories
+
+* As an Site Reliability Engineer managing multiple clusters, I'd like to run compliance
+  scans on my fleet in such a way that uses the minimal amount of resources.
+
+* As a Site Reliability Engineer managing multiple clusters, I'd like to have
+  one place to view the compliance stance of my whole fleet.
+
+### API Extensions
+
+Given the scope of the change, we'd upgrade the ScanSetting's version to `v1alpha2`.
+
+A sample `ScanSetting` object with Compliance Operator's current mode of operation would look as follows:
+
+```yaml
+apiVersion: compliance.openshift.io/v1alpha2
```

Comment:
I tend to agree that alpha2 is better for now, but if the design proves to be stable, let's go beta..